Tests/TuistDependenciesTests/SwiftPackageManager/Utils/SwiftPackageManagerGraphGeneratorTests.swift

```diff
@@ -0,0 +1,76 @@
+import TuistDependencies
+import TuistGraph
+import XCTest
+@testable import TuistDependenciesTesting
+@testable import TuistSupportTesting
+
+class SwiftPackageManagerGraphGeneratorTests: TuistTestCase {
+    private var swiftPackageManagerController: MockSwiftPackageManagerController!
+    private var subject: SwiftPackageManagerGraphGenerator!
+
+    override func setUp() {
+        super.setUp()
+
+        swiftPackageManagerController = MockSwiftPackageManagerController()
+        subject = SwiftPackageManagerGraphGenerator(swiftPackageManagerController: swiftPackageManagerController)
+    }
+
+    override func tearDown() {
+        super.tearDown()
+
+        fileHandler = nil
+        swiftPackageManagerController = nil
+        subject = nil
+    }
+
+    func test_generate() throws {
+        // Given
+        let path = try temporaryPath()
+        let checkoutsPath = path.appending(component: "checkouts")
+
+        fileHandler.stubContentsOfDirectory = { path in
+            XCTAssertEqual(path, checkoutsPath)
+            return [
+                checkoutsPath.appending(component: "alamofire"),
+                checkoutsPath.appending(component: "google-app-measurement"),
+            ]
+        }
+
+        var loadPackageInfoCalls = 0
+        swiftPackageManagerController.loadPackageInfoStub = { path in
+            loadPackageInfoCalls += 1
+            switch loadPackageInfoCalls {
```

Comment:
I'd probably differentiate which `PackageInfo` to return based on the path. Something along the lines of:
```swift
if path == checkoutsPath.appending(component: "google-app-measurement").appending(component: "Package.swift") {
  // return
} else {
  XCTFail("Unexpected path")
}
```